File: app/src/main/java/it/unical/mat/embasp/languages/asp/ASPInputProgram.java
package it.unical.mat.embasp.languages.asp;

import java.util.Set;

import it.unical.mat.embasp.base.InputProgram;

/** a generic ASP program , with the capabilities of retrieve data by objects */

public class ASPInputProgram extends InputProgram {

	public ASPInputProgram() {
		super();
	}

	public ASPInputProgram(final Object inputObj) throws Exception {
		super(inputObj);
	}

	public ASPInputProgram(final String initial_program) {
		super(initial_program);
	}

	/**
	 * transforms a given Object class into a {@link InputProgram} and adds it to the current {@link #programs}
	 *
	 * @param inputObj
	 *            an object to be transformed
	 * @see ASPMapper
	 * @throws IllegalAccessException
	 *             , IllegalArgumentException , InvocationTargetException , NoSuchMethodException , SecurityException , IllegalTermException,
	 *             ObjectNotValidException
	 */
	@Override
	public void addObjectInput(final Object inputObj) throws Exception {
		addProgram(ASPMapper.getInstance().getString(inputObj) + ".");
	}

	/** transforms a set of objects @see #addObjectInput(Object) */
	@Override
	public void addObjectsInput(final Set<Object> inputObjs) throws Exception {
		for (final Object inputObj : inputObjs)
			addObjectInput(inputObj);
	}
}

File: Python version/languages/asp/ASPInputProgram.py
from base.InputProgram import InputProgram
from languages.asp.ASPMapper import ASPMapper

class ASPInputProgram(InputProgram):
    """A generic ASP program, with the capabilities of retrieve data by objects"""
    
    def __init__(self):
        super(ASPInputProgram, self).__init__()
    
    
    def addObjectInput(self, inputObj):
        """Transforms a given Object class into a InputProgram and adds it to the current _programs
        The parameter inputObj is an object to be transformed
        """
        self.addProgram(ASPMapper.getInstance().getString(inputObj) + ".")
    
    
    def addObjectsInput(self, inputObjs):
        """Transforms a set of objects"""
        for inputObj in inputObjs:
            self.addObjectInput(inputObj)